nucypher/blockchain/eth/signers.py

```diff
@@ -191,3 +216,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+
+    IS_SIGNER_LOCAL = True
+
+    def __init__(self, uri: str):
+        super().__init__()
+
+        decoded_uri = urlparse(uri)
+        if decoded_uri.scheme not in ('key', 'keystore') or decoded_uri.netloc:
+            raise cls.InvalidSignerURI(uri)
+
+        with open(decoded_uri.path) as file_obj:
+            key = json.load(file_obj)
+
+        address = key['address']
+        if not Web3.isAddress(address):
+            raise ValueError(f"'{decoded_uri.path}' does not contain a valid address")
+
+        self.key = key
+        self._account = Web3.toChecksumAddress(address)
+        self._signer = None
+
+    def _is_my_account(self, account) -> bool:
+        return Web3.isAddress(account) and \
+            self.account == Web3.toChecksumAddress(account)
+
+    @classmethod
+    def from_signer_uri(cls, uri: str) -> 'Signer':
+        return cls(uri)
```

Comment:
Hmm - Just a passthrough method here?  Just a suggestion: Parse the URI itself here, then passing the keyfile or password into the `__init__`.  This starts to suggest that this class is more aptly named `KeyfileSigner`, `LocalPrivateKeySigner`, etc.

Reply:
Done - URI decoded has been factored out as suggested.